sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -53,54 +52,49 @@ public static AmqpMessage BatchSBMessagesAsAmqpMessage(IReadOnlyCollection<Servi
         ///
         /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
         ///
-        private static AmqpMessage BuildAmqpBatchFromMessage(IReadOnlyCollection<ServiceBusMessage> source, bool forceBatch)
+        public virtual AmqpMessage BuildAmqpBatchFromMessage(IReadOnlyCollection<ServiceBusMessage> source, bool forceBatch)
         {
-            AmqpMessage firstAmqpMessage = null;
-            ServiceBusMessage firstMessage = null;
-
             var batchMessages = new List<AmqpMessage>(source.Count);
             foreach (ServiceBusMessage sbMessage in source)
             {
-                if (firstAmqpMessage == null)
-                {
-                    firstAmqpMessage = SBMessageToAmqpMessage(sbMessage);
-                    firstMessage = sbMessage;
-                    batchMessages.Add(firstAmqpMessage);
-                }
-                else
-                {
-                    batchMessages.Add(SBMessageToAmqpMessage(sbMessage));
-                }
+                 batchMessages.Add(SBMessageToAmqpMessage(sbMessage));
             }
-
-            return BuildAmqpBatchFromMessages(batchMessages, firstMessage, forceBatch);
+            return BuildAmqpBatchFromMessages(batchMessages, forceBatch);
         }
 
         /// <summary>
         ///   Builds a batch <see cref="AmqpMessage" /> from a set of <see cref="AmqpMessage" />.
         /// </summary>
         ///
-        /// <param name="batchMessages">The set of messages to use as the body of the batch message.</param>
-        /// <param name="firstMessage">The first message being sent in the batch.</param>
+        /// <param name="source">The set of messages to use as the body of the batch message.</param>
         /// <param name="forceBatch">Set to true to force creating as a batch even when only one message.</param>
         ///
         /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
         ///
-        private static AmqpMessage BuildAmqpBatchFromMessages(
-            List<AmqpMessage> batchMessages,
-            ServiceBusMessage firstMessage,
+        public virtual AmqpMessage BuildAmqpBatchFromMessages(
```

Comment:
Does this need to be public? Can we instead call into the wrapper?

Reply:
The reason I made this public was just because I needed a way to pass the AMQP messages around instead of the Service Bus Messages. I think it could work to add a wrapper that takes in a `ServiceBusMessageBatch` and then make this method internal, would that be better?

Reply:
But otherwise, the only other wrappers just take 1+ `ServiceBusMessage`'s

Reply:
Ah I see. I guess we do need this one to be public. But what about the other one that takes a ServiceBusMessage - https://github.com/Azure/azure-sdk-for-net/pull/30305/files/16bad73f62f1503c888b6a068d214c316a0f8198#diff-a71c7b0a7a2b2fe832ac51ebab6b7aff0f760f124d542984cde6f7b8e566b9f8R55

Reply:
That one can be private. It just can't be static

Reply:
That's fine if it is not static.